docs/remote-access/wireguard.md

```diff
@@ -11,4 +11,112 @@ This page will cover two WireGuard architectures. The first is running WireGuard
 
 ## WireGuard Server on a VPS
 
-* Todo
\ No newline at end of file
+This section will talk about setting up a VPS and then running a client on your network to setup the connection to the VPS and allow the vpn to work without a firewall opening. 
+
+The first step is to decide on a virtual host provider and create a server. When you decide on a provider and create a host. Install Wireguard using the instructions provided on the [Wireguards Documentation](https://www.wireguard.com/install/)
+
+### Server Configuration Setup:
+First create a folder to contain all the details of our configuration.
+```shell
+mkdir wireguard
+cd wireguard
+```
+
+Now you must create the keys you have to options on how many keys you need to make. The simplest allows for just two keys but wireguard also allows you to provide a preshared key as well. For this configuration we will work with the preshared key but if you do not want to you can remove those files from the configuration.
+
+```shell
+wg genkey | tee cloud_key.priv | wg pubkey > cloud_key.pub
+wg genkey | tee client_1_key.priv | wg pubkey > client_1_key.pub
+wg genpsk > client_1_cloud_preshared_key.psk
+```
+
+```shell
+vim /etc/wireguard/wg0.conf
+```
+
+When you create the file you will copy the following configuration:
+```
+[Interface]
+# Cloud
+Address = 192.168.1.1/24
+ListenPort = 51902
+PrivateKey = << Private Key - cloud_key.priv >>
+
+PostUp = iptables -A FORWARD -i %i -j ACCEPT ; iptables -A FORWARD -o %i -j ACCEPT; iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
```

Comment:
These `iptables` commands aren't necessary for a plain tunnel. They're more useful when doing site-to-site tunnels.

Reply:
Loads of WireGuard guides just include them and spam these things without explaining what they are for. A dedicated section would be ideal.

Reply:
> The last two items in the Interface definition are the commands that will be run after the interface is stood up or taken down. The commands included in the provided configuration allows anything put over the wireguard network to this interface to be forwarded to the network it is on.

I included this blurb about it. I like the idea of breaking this out, into its own section for site to site. This would also help with a comment below. about the second IP. I included a description below each configuration file, to help explain it. This is something some tutorials I think didnt explain it well for myself at least.  I will work on an update, to take this into account.